src/server/test/hotspot_partition_test.cpp

```diff
@@ -22,27 +22,108 @@
-    test_rows[6].get_qps = 1000.0;
-    test_rows[7].get_qps = 5000.0;
     hotspot_partition_calculator test_hotspot_calculator("TEST", 8);
-    test_hotspot_calculator.data_aggregate(test_rows);
-    test_hotspot_calculator.data_analyse();
-    std::vector<double> result(8);
-    for (int i = 0; i < test_hotspot_calculator._hot_points.size(); i++) {
-        result[i] = test_hotspot_calculator._hot_points[i]->get_value();
-    }
-    std::vector<double> expect_vector{0, 0, 0, 0, 0, 0, 0, 3};
-    ASSERT_EQ(expect_vector, result);
+    {
+        // Insert normal scenario data to test
+        test_hotspot_calculator.data_aggregate(std::move(generate_normal_scenario_0()));
+        test_hotspot_calculator.data_analyse();
+        std::vector<std::vector<double>> result =
+            get_calculator_result(test_hotspot_calculator._hot_points);
+        std::vector<double> read_expect_vector{0, 0, 0, 0, 0, 0, 0, 0};
+        std::vector<double> write_expect_vector{0, 0, 0, 0, 0, 0, 0, 0};
+        ASSERT_EQ(result[READ_HOTSPOT_DATA], read_expect_vector);
+        ASSERT_EQ(result[WRITE_HOTSPOT_DATA], write_expect_vector);
+    }
+
+    {
+        // Insert hot scenario 0 data to test
+        test_hotspot_calculator.data_aggregate(std::move(generate_hot_scenario_0()));
+        test_hotspot_calculator.data_analyse();
+        std::vector<std::vector<double>> result =
+            get_calculator_result(test_hotspot_calculator._hot_points);
+        std::vector<double> read_expect_vector{0, 0, 0, 0, 0, 0, 0, 4};
+        std::vector<double> write_expect_vector{4, 0, 0, 0, 0, 0, 0, 0};
+        ASSERT_EQ(result[READ_HOTSPOT_DATA], read_expect_vector);
+        ASSERT_EQ(result[WRITE_HOTSPOT_DATA], write_expect_vector);
+    }
+
+    {
+        // Insert hot scenario 0 data to test again
+        test_hotspot_calculator.data_aggregate(std::move(generate_hot_scenario_0()));
+        test_hotspot_calculator.data_analyse();
+        std::vector<std::vector<double>> result =
+            get_calculator_result(test_hotspot_calculator._hot_points);
+        std::vector<double> read_expect_vector{0, 0, 0, 0, 0, 0, 0, 4};
+        std::vector<double> write_expect_vector{4, 0, 0, 0, 0, 0, 0, 0};
+        ASSERT_EQ(result[READ_HOTSPOT_DATA], read_expect_vector);
+        ASSERT_EQ(result[WRITE_HOTSPOT_DATA], write_expect_vector);
+    }
+
+    {
+        // Insert hot scenario 1 data to test
+        test_hotspot_calculator.data_aggregate(std::move(generate_hot_scenario_1()));
+        test_hotspot_calculator.data_analyse();
+        std::vector<std::vector<double>> result =
+            get_calculator_result(test_hotspot_calculator._hot_points);
+        std::vector<double> read_expect_vector{0, 0, 0, 4, 0, 0, 0, 0};
+        std::vector<double> write_expect_vector{0, 0, 4, 0, 0, 0, 0, 0};
+        ASSERT_EQ(result[READ_HOTSPOT_DATA], read_expect_vector);
+        ASSERT_EQ(result[WRITE_HOTSPOT_DATA], write_expect_vector);
+    }
```

Comment:
refactor to one method: test_XXX(std::vector<row_data> scenario, std::vector<double> read_expect_vector,  std::vector<double> write_expect_vector)